orchagent/aclorch.cpp

```diff
@@ -1354,9 +1365,22 @@ bool AclTable::create()
     attr.value.booldata = true;
     table_attrs.push_back(attr);
 
-    attr.id = SAI_ACL_TABLE_ATTR_FIELD_IP_PROTOCOL;
-    attr.value.booldata = true;
-    table_attrs.push_back(attr);
+    // NOTE: Temporary workaround to support matching protocol numbers on MLNX platform.
+    // In a later SAI version we will transition to using NEXT_HEADER for IPv6 on all platforms.
+    string platform = getenv("platform") ? getenv("platform") : "";
+    if ((type == ACL_TABLE_MIRRORV6 || type == ACL_TABLE_L3V6)
+            && platform == MLNX_PLATFORM_SUBSTRING)
```

Comment:
>MLNX_PLATFORM_SUBSTRING [](start = 27, length = 23)

SUBSTRING means to test if it is substring of a platform string.

Reply:
I see it used in this `==` manner throughout aclorch and orchdaemon. Maybe the macro names should be updated?

Reply:
I see it already mixed used. Let's keep as-is, and add an issue to completely fix for all platform string comparison.

---
In reply to: [450408833](https://github.com/Azure/sonic-swss/pull/1343#discussion_r450408833) [](ancestors = 450408833)

Reply:
Opened #1344. Thanks, Qi!